File: test-tools.js
#!/usr/bin/env node

const { execFile } = require('child_process');
const { promisify } = require('util');
const execFileAsync = promisify(execFile);

async function testTool(toolName) {
    console.log(`\n🔍 Testing ${toolName}...`);
    
    try {
        const { stdout, stderr } = await execFileAsync('python3', ['-m', toolName, '--help'], {
            timeout: 10000,
            maxBuffer: 1024 * 1024,
            env: {
                ...process.env,
                PYTHONUNBUFFERED: '1',
                PYTHONIOENCODING: 'utf-8',
                PYTHONUTF8: '1',
                LC_ALL: 'C.UTF-8',
                LANG: 'C.UTF-8',
                LANGUAGE: 'C.UTF-8',
                TERM: 'dumb',
                NO_COLOR: '1',
                FORCE_COLOR: '0',
                ANSI_COLORS_DISABLED: '1',
                CLICOLOR: '0',
                CLICOLOR_FORCE: '0'
            }
        });
        
        console.log(`✅ ${toolName} is working!`);
        console.log(`📤 stdout length: ${stdout?.length || 0}`);
        console.log(`📤 stderr length: ${stderr?.length || 0}`);
        
        if (stdout && stdout.length > 0) {
            console.log(`📄 ${toolName} help preview:`, stdout.substring(0, 200) + '...');
        }
        
        return true;
    } catch (error) {
        console.log(`❌ ${toolName} failed:`, error.message);
        return false;
    }
}

async function main() {
    console.log('🧪 Testing Python OSINT Tools...');
    console.log('🔧 Environment:', process.env.NODE_ENV || 'development');
    console.log('🐍 Python path:', process.env.PYTHON_PATH || 'python3');
    
    const tools = ['sherlock', 'holehe', 'maigret', 'ghunt'];
    const results = [];
    
    for (const tool of tools) {
        const success = await testTool(tool);
        results.push({ tool, success });
    }
    
    console.log('\n📊 Test Results:');
    results.forEach(({ tool, success }) => {
        console.log(`   ${success ? '✅' : '❌'} ${tool}`);
    });
    
    const allWorking = results.every(r => r.success);
    console.log(`\n${allWorking ? '🎉 All tools are working!' : '⚠️ Some tools failed. Check the output above.'}`);
    
    if (!allWorking) {
        console.log('\n🔧 Troubleshooting tips:');
        console.log('1. Make sure Python 3 is installed');
        console.log('2. Check if tools are installed: pip3 list | grep -E "(sherlock|holehe|maigret|ghunt)"');
        console.log('3. Try installing tools: pip3 install sherlock-project maigret holehe ghunt');
        console.log('4. Check Python path: which python3');
    }
}

main().catch(console.error);

File: test_tools.py
#!/usr/bin/env python3
import subprocess
import os
import sys

def test_tool(tool, args):
    print(f"\n🔍 Testing {tool} with args: {args}")
    
    # Try different execution methods
    methods = [
        # Method 1: Direct execution
        lambda: subprocess.run([tool] + args, capture_output=True, text=True, timeout=30),
        # Method 2: pipx run
        lambda: subprocess.run(["pipx", "run", tool] + args, capture_output=True, text=True, timeout=30),
        # Method 3: python -m
        lambda: subprocess.run(["python3", "-m", tool] + args, capture_output=True, text=True, timeout=30),
        # Method 4: Shell with PATH
        lambda: subprocess.run(f"export PATH='/opt/render/.local/bin:/home/render/.local/bin:/usr/local/bin:/usr/bin:/bin:$PATH' && {tool} {' '.join(args)}", shell=True, capture_output=True, text=True, timeout=30)
    ]
    
    for i, method in enumerate(methods, 1):
        try:
            print(f"  Method {i}: Trying {method.__name__}...")
            result = method()
            print(f"    Return code: {result.returncode}")
            print(f"    Stdout: {result.stdout[:200]}...")
            print(f"    Stderr: {result.stderr[:200]}...")
            if result.returncode == 0 or result.stdout.strip():
                print(f"    ✅ SUCCESS with method {i}")
                return result
        except Exception as e:
            print(f"    ❌ Method {i} failed: {e}")
    
    print(f"  ❌ All methods failed for {tool}")
    return None

if __name__ == "__main__":
    print("🧪 Testing OSINT Tools...")
    
    # Test each tool
    tools_to_test = [
        ("holehe", ["aikyanaskar2006@gmail.com"]),
        ("ghunt", ["email", "aikyanaskar2006@gmail.com"]),
        ("sherlock", ["aikyanaskar2006"]),
        ("maigret", ["aikyanaskar2006"])
    ]
    
    for tool, args in tools_to_test:
        result = test_tool(tool, args)
        if result:
            print(f"✅ {tool} working!")
        else:
            print(f"❌ {tool} not working!")
    
    print("\n🔍 Checking available tools in PATH...")
    try:
        result = subprocess.run(["which", "holehe", "ghunt", "sherlock", "maigret"], capture_output=True, text=True)
        print(f"Available tools: {result.stdout}")
    except:
        print("Could not check which tools are available")
    
    print("\n🔍 Checking pipx installations...")
    try:
        result = subprocess.run(["pipx", "list"], capture_output=True, text=True)
        print(f"Pipx installations: {result.stdout}")
    except:
        print("Could not check pipx installations")
